datavec/datavec-python/src/main/java/org/datavec/python/NumpyArray.java

```diff
@@ -0,0 +1,114 @@
+package org.datavec.python;
+
+import org.apache.commons.lang3.ArrayUtils;
+import org.bytedeco.javacpp.Pointer;
+import org.json.simple.JSONArray;
+import org.json.simple.JSONObject;
+import org.nd4j.linalg.api.buffer.DataBuffer;
+import org.nd4j.linalg.api.buffer.util.DataTypeUtil;
+import org.nd4j.linalg.api.ndarray.INDArray;
+import org.bytedeco.javacpp.FloatPointer;
+import org.bytedeco.javacpp.DoublePointer;
+import org.nd4j.linalg.dataset.DataSet;
+import org.nd4j.linalg.factory.Nd4j;
+import org.nd4j.nativeblas.NativeOps;
+import org.nd4j.nativeblas.NativeOpsHolder;
+import org.nd4j.linalg.api.buffer.DataType;
+
+
+import java.util.ArrayList;
+
+
+public class NumpyArray {
+
+    private static NativeOps nativeOps = NativeOpsHolder.getInstance().getDeviceNativeOps();
+    private long address;
+    private long[] shape;
+    private long[] strides;
+    private DataType dtype = DataType.FLOAT;
+    private INDArray nd4jArray;
+
+    public NumpyArray(long address, long[] shape, long strides[], boolean copy){
+        this.address = address;
+        this.shape = shape;
+        this.strides = strides;
+        setND4JArray();
+        if (copy){
+            nd4jArray = nd4jArray.dup();
+            this.address = nd4jArray.data().address();
+
+        }
+    }
+    public NumpyArray(long address, long[] shape, long strides[]){
+        this(address, shape, strides, false);
+    }
+
+    public NumpyArray(long address, long[] shape, long strides[], DataType dtype){
+        this(address, shape, strides, dtype, false);
+    }
+
+    public NumpyArray(long address, long[] shape, long strides[], DataType dtype, boolean copy){
+        this.address = address;
+        this.shape = shape;
+        this.strides = strides;
+        this.dtype = dtype;
+        setND4JArray();
+        if (copy){
+            nd4jArray = nd4jArray.dup();
+            this.address = nd4jArray.data().address();
+        }
+    }
+
+    public long getAddress() {
+        return address;
+    }
+
+    public long[] getShape() {
+        return shape;
+    }
+
+    public long[] getStrides() {
+        return strides;
+    }
+
+    public DataType getDType() {
+        return dtype;
+    }
+
+    public INDArray getND4JArray() {
+        return nd4jArray;
+    }
+
+    private void setND4JArray(){
+        long size = 1;
+        for(long d: shape){
+            size *= d;
+        }
+        Pointer ptr = nativeOps.pointerForAddress(address);
+        ptr = ptr.limit(size);
```

Comment:
@saudet mind checking this part?

Reply:
Now that you mention it, the size needs to take into account the element size somehow because this is a raw buffer, although this might be reset somewhere in Nd4j.createBuffer() already, but it looks like that's missing for the new types.

Reply:
The tests pass. We have tests for float, double, int, short and long. So it works right now.

Reply:
It probably works for most things, but we're going to hit some place where it won't work. Anyway, this is actually something that should be fixed in https://github.com/deeplearning4j/deeplearning4j/blob/master/nd4j/nd4j-buffer/src/main/java/org/nd4j/linalg/api/buffer/factory/DefaultDataBufferFactory.java#L566  where those values should be set in the `create(Pointer, ...)` method like they are in `create(DoublePointer ...)`, etc.

Reply:
@farizrahman4u can you resolve this if this concern is fixed?